processor/deltatocumulativeprocessor/internal/delta/delta.go

```diff
@@ -0,0 +1,61 @@
+package delta
+
+import (
+	"fmt"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/deltatocumulativeprocessor/internal/data"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/deltatocumulativeprocessor/internal/streams"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+)
+
+func construct[D data.Point[D]]() streams.Aggregator[D] {
+	acc := &Accumulator[D]{dps: make(map[streams.Ident]D)}
+	lock := &Lock[D]{next: acc}
+	return lock
```

Comment:
[Nit] Why not just return the instantiated object directly? I don't see the point of a variable:
```suggestion
	return &Lock[D]{next: acc}
```